device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceTwin/DeviceTwin.java

```diff
@@ -153,8 +156,11 @@ private void OnDesiredPropertyChanged(TwinCollection desiredPropertyMap)
         {
             if (desiredPropertyMap != null)
             {
-                if ( onDesiredPropertyChangeMap != null ) {
```

Comment:
I didn't follow exactly what you meant with this if block, so I made it so that if the user provides a ```TwinPropertiesCallback``` implementation, our library will execute that callback whenever a desired property update arrives with at least one property. Is that the behavior your want?